Fix CI for Zarr Python v3

The [build is failing](https://github.com/sgkit-dev/sgkit/actions/runs/12738782977/job/35501589696) with 

```
FAILED sgkit/tests/io/plink/test_plink_writer.py::test_zarr_to_plink[plink_sim_10s_100v_10pmiss-\t] - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
FAILED sgkit/tests/io/plink/test_plink_writer.py::test_zarr_to_plink[example- ] - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
FAILED sgkit/tests/io/plink/test_plink_writer.py::test_zarr_to_plink[example_with_fam- ] - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
FAILED sgkit/tests/io/test_dataset.py::test_save_and_load_dataset[True] - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
FAILED sgkit/tests/io/test_dataset.py::test_save_and_load_dataset[False] - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
FAILED sgkit/tests/io/test_dataset.py::test_save_and_load_dataset__mutable_mapping - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
FAILED sgkit/tests/test_association.py::test_regenie_loco_regression[True-numpy] - AttributeError: 'Array' object has no attribute 'compressors'
FAILED sgkit/tests/test_association.py::test_regenie_loco_regression[False-numpy] - AttributeError: 'Array' object has no attribute 'compressors'
FAILED sgkit/tests/test_regenie.py::test_regenie__glow_comparison[numpy] - AttributeError: 'Array' object has no attribute 'compressors'
FAILED sgkit/tests/test_testing.py::test_simulate_genotype_call_dataset__zarr - TypeError: Group.create_array() got an unexpected keyword argument 'overwrite'
```

We need to update to use the 3.0.0 release, and possibly a new release of bio2zarr.

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -43,7 +43,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        zarr: ["==3.0.0b1"]
+        zarr: [">=3"]
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-python@v5

--- a/docs/examples/1kg.schema.json
+++ b/docs/examples/1kg.schema.json
@@ -1369,6 +1369,29 @@
             "filters": []
         },
         {
+            "name": "variant_length",
+            "dtype": "i1",
+            "shape": [
+                10879
+            ],
+            "chunks": [
+                10000
+            ],
+            "dimensions": [
+                "variants"
+            ],
+            "description": "The length of the variant measured in bases",
+            "vcf_field": "rlen",
+            "compressor": {
+                "id": "blosc",
+                "cname": "zstd",
+                "clevel": 7,
+                "shuffle": 0,
+                "blocksize": 0
+            },
+            "filters": []
+        },
+        {
             "name": "call_AD",
             "dtype": "i1",
             "shape": [

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -13,7 +13,7 @@ partd
 bed-reader
 rechunker
 cbgen < 1.0.5
-bio2zarr; platform_system != "Windows"
+bio2zarr @ git+https://github.com/sgkit-dev/bio2zarr.git; platform_system != "Windows"
 yarl
 matplotlib
 asv

--- a/sgkit/tests/io/test_dataset.py
+++ b/sgkit/tests/io/test_dataset.py
@@ -58,7 +58,7 @@ def test_save_unequal_chunks_error():
     # Normal zarr errors shouldn't be caught
     with pytest.raises(
         (FileExistsError, ValueError),
-        match="(path '' contains an array|Store already exists)",
+        match="(path '' contains an array|is not empty)",
     ):
         save_dataset(ds, {".zarray": ""})
 

--- a/sgkit/tests/test_testing.py
+++ b/sgkit/tests/test_testing.py
@@ -7,6 +7,9 @@ import xarray as xr
 from sgkit.testing import simulate_genotype_call_dataset
 
 
+@pytest.mark.filterwarnings(
+    "ignore::UserWarning"
+)  # codec `vlen-utf8` not in Zarr v3 spec`
 def test_simulate_genotype_call_dataset__zarr(tmp_path):
     path = str(tmp_path / "ds.zarr")
     ds = simulate_genotype_call_dataset(n_variant=10, n_sample=10)

